src/main/java/eu/darkbot/api/core/Memory.java

```diff
@@ -0,0 +1,276 @@
+package eu.darkbot.api.core;
+
+/**
+ * This interface provides API to read/write native memory
+ * <p>
+ * Those calls should be generally safe,
+ * every access violation error is handled by native code
+ */
+public interface Memory {
+
+    /**
+     * Reads signed integer value from memory.
+     *
+     * @param address to read
+     * @return signed integer value
+     */
+    int readInt(long address);
+
+    default int readInt(long address, int... offsets) {
+        int i = 0;
+        for (; i < offsets.length - 1; i++)
+            address = readLong(address + offsets[i]);
+
+        return readInt(address + offsets[i]);
+    }
+
+    /**
+     * Reads signed long value from memory.
+     *
+     * @param address to read
+     * @return signed long value
+     */
+    long readLong(long address);
+
+    default long readLong(long address, int... offsets) {
+        for (int offset : offsets)
+            address = readLong(address + offset);
+
+        return address;
+    }
+
+    /**
+     * Reads double value from memory.
+     *
+     * @param address to read
+     * @return double value
+     */
+    double readDouble(long address);
+
+    default double readDouble(long address, int... offsets) {
+        int i = 0;
+        for (; i < offsets.length - 1; i++)
+            address = readLong(address + offsets[i]);
+
+        return readDouble(address + offsets[i]);
+    }
+
+    /**
+     * Reads boolean value from memory.
+     *
+     * @param address to read
+     * @return boolean value
+     */
+    boolean readBoolean(long address);
+
+    default boolean readBoolean(long address, int... offsets) {
+        int i = 0;
+        for (; i < offsets.length - 1; i++)
+            address = readLong(address + offsets[i]);
+
+        return readBoolean(address + offsets[i]);
+    }
+
+    /**
+     * Reads String from memory.
+     *
+     * @param address to read
+     */
+    String readString(long address);
+
+    default String readString(long address, int offsets) {
```

Comment:
```suggestion
    default String readString(long address, int... offsets) {
```
That's supposed to be an array right?